core/src/wsv.rs

```diff
@@ -684,6 +684,34 @@ impl<W: WorldTrait> WorldStateView<W> {
         self.new_block_notifier.subscribe()
     }
 
+    /// Get all transactions
+    pub fn transaction_values(&self) -> Vec<TransactionValue> {
```

Comment:
I think transactions should be grouped by block number, i.e. function prototype should be something like this:
```suggestion
    pub fn transaction_values(&self) -> Vec<(u32, Vec<TransactionValue>)> {
```

or block number can be implicit in the position in the vector:
```suggestion
    pub fn transaction_values(&self) -> Vec<Vec<TransactionValue>> {
```

I somehow dislike the idea of mixing valid and rejected transactions so maybe this format could be favored:
 ```suggestion
    pub fn transaction_values(&self) -> Vec<(Vec<ValidTransaction>, Vec<RejectedTransaction>)> {
```

Reply:
Oh, this function isn't actually needed anymore since I split the transactions into valid and rejected, so I should actually remove it.

That said, having block numbers in `FindAllTransactions` output makes sense to me, but again, not sure what the actual use-cases of the query are.

Reply:
that's true, I too don't know the use-case of this feature so you could just open a ticket for this